### PR TITLE
[TNS - 734, 735, 736] - Deck creation & shuffle

### PR DESCRIPTION
**Pull request by rodrigoaadias**
- All _must have_ cards type created.
- Deck is created by `Match `class with 104 cards
- Deck cards is shuffled before distribute card to player
- Each player receive 7 random cards to play

### CODE REVIEW

File: Uno-App/Source/Private/Match.cpp [hunk 1]
```diff
@@ -17,13 +19,15 @@ void Match::StartNewMatch()
...
+    CreateDeck();
+    SortCardsToPlayers();
 }
 
 void Match::JoinPlayers(const int& number)
 {
     for (int i=0; i < number; i++)
     {
-        JoinedPlayers.push_back(CreatePlayer(i));
+        JoinedPlayers.emplace_back(CreatePlayer(i));
```
Comment (Felipessoaf) [inline Uno-App/Source/Private/Match.cpp:32]: Why did you change this?

Comment (rodrigoaadias) [reply Uno-App/Source/Private/Match.cpp:32]: I watched a video from Cherno and he said that `push_back` copy the value to the list and `emplace` add its reference directly without copying it. It's a performance improvement

**Review by Felipessoaf — APPROVED** [review]
